Integration Monitor (#5564)
DataDog/documentation: content/en/monitors/monitor_types/integration.md

  text: "Schedule a downtime to mute a monitor"
- link: "monitors/monitor_status"
  tag: "Documentation"
  text: "Check your monitor status"
---

## Overview

Use an integration monitor to check if an installed integration is running. For more detailed monitoring, a metric monitor can be used to gauge specific information about an integration.

## Monitor creation

To create an [integration monitor][1] in Datadog:

1. Use the main navigation: *Monitors --> New Monitor --> Integration*.
2. Search for an integration or select it from the list or images.
3. Choose an **Integration Metric** or **Integration Status** monitor:
    {{< img src="monitors/monitor_types/integration/metric_or_status.png" alt="Metric or Status" responsive="true" style="width:90%;">}}

### Integration Metric

> Create an integration metric monitor by following the instructions in the [metric monitor][2] documentation.

Reviewer: ```suggestion
Create an integration metric monitor by following the instructions in the [metric monitor][2] documentation.  Creating an integration metric monitor ensures that the monitor will be selected by the Integration Monitor Type facet on the Manage Monitors page
```

### Integration Status

> If the integration has a status metric, the **Integration Status** tab is selectable.

Reviewer: ```suggestion
If the integration has a service check, the **Integration Status** tab is selectable.
```

#### Pick a check

If there is only one check for the integration, no selection is necessary. Otherwise, select the check for your monitor.

#### Pick monitor scope

Select the scope to monitor by choosing host names, tags, or choose `All Monitored Hosts`. If you need to exclude certain hosts, use the second field to list names or tags.

* The include field uses `AND` logic. All listed names and tags must be present on a host for it to be included.
* The exclude field uses `OR` logic. Any host with a listed name or tag is excluded.

#### Set alert conditions

In this section, choose between a **Check Alert** or **Cluster Alert**:

{{< tabs >}}
{{% tab "Check Alert" %}}

A check alert tracks consecutive statuses submitted per check grouping and compares it to your thresholds.

Set up the check alert:

1. Trigger a separate alert for each `<GROUP>` reporting your check.

    Check grouping is specified either from a list of known groupings or by you. For integration monitors, the per-check grouping is explicitly known. For example, the Postgres integration is tagged with `db`, `host`, and `port`.
    
2. Trigger the alert after selected consecutive failures: `<NUMBER>`

>     Each check run submits a single status of `OK`, `WARN`, or `CRITICAL`. Choose how many consecutive runs with the `CRITICAL` status trigger a notification. For example, your database might have a single blip where connection fails. If you set this value to `> 1`, the blip is ignored but a problem with more than one failure triggers a notification.

Reviewer: ```suggestion
    Each check run submits a single status of `OK`, `WARN`, `CRITICAL`, or `UNKNOWN`. Choose how many consecutive runs with the `CRITICAL` status trigger a notification. For example, your database might have a single blip where connection fails. If you set this value to `> 1`, the blip is ignored but a problem with more than one consecutive failure triggers a notification.
```

  text: "Schedule a downtime to mute a monitor"
- link: "monitors/monitor_status"
  tag: "Documentation"
  text: "Check your monitor status"
---

## Overview

> Use an integration monitor to check if an installed integration is running. For more detailed monitoring, a metric monitor can be used to gauge specific information about an integration.

Reviewer 2: ```suggestion
Use an integration monitor to check if an installed [integration](/integrations/) is running. For more detailed monitoring, a [metric monitor](/monitors/monitor_types/metric/) can be used to gauge specific information about an integration.
```

Author: Added the link to integrations but adding the link to metric monitors here doesn't seem appropriate.

## Monitor creation

To create an [integration monitor][1] in Datadog:

1. Use the main navigation: *Monitors --> New Monitor --> Integration*.
2. Search for an integration or select it from the list or images.
3. Choose an **Integration Metric** or **Integration Status** monitor:
    {{< img src="monitors/monitor_types/integration/metric_or_status.png" alt="Metric or Status" responsive="true" style="width:90%;">}}

### Integration Metric

Create an integration metric monitor by following the instructions in the [metric monitor][2] documentation. Using the integration metric monitor type ensures the monitor can be selected by the integration monitor type facet on the [Manage Monitors][3] page.

### Integration Status

> If the integration has a service check, the **Integration Status** tab is selectable.

Reviewer 2: ```suggestion
If the integration has a service check, the **Integration Status** tab is active.
```

> #### Pick a check

Reviewer 2: Should we create a numbered list here since it's a set of instructions ? instead of using headers?

Author: Nope. I've established a template where all headers in monitor instructions are headers in the docs. This makes it easy for support to link to a particular step.

If there is only one check for the integration, no selection is necessary. Otherwise, select the check for your monitor.

#### Pick monitor scope

> Select the scope to monitor by choosing host names, tags, or choose `All Monitored Hosts`. If you need to exclude certain hosts, use the second field to list names or tags.

Reviewer 2: ```suggestion
Select the scope to monitor by choosing host names, tags, or choose `All Monitored Hosts`. If you need to exclude certain hosts, use the second field to list names or tags for the host to remove.
```

Author: I don't think a colon is appropriate here.

> * The include field uses `AND` logic. All listed names and tags must be present on a host for it to be included.

Reviewer 2: ```suggestion
* The include field uses `AND` logic. All listed host names and tags must be present on a host for it to be included.
```

> * The exclude field uses `OR` logic. Any host with a listed name or tag is excluded.

Reviewer 2: ```suggestion
* The exclude field uses `OR` logic. Any host with a listed host name or tag is excluded.
```

#### Set alert conditions

In this section, choose between a **Check Alert** or **Cluster Alert**:

{{< tabs >}}
{{% tab "Check Alert" %}}

> A check alert tracks consecutive statuses submitted per check grouping and compares it to your thresholds.

Reviewer 2: ```suggestion
A check alert tracks consecutive statuses submitted per check and compares it to your thresholds.
```
